opengovsg/postmangovsg
# chore(lottie): migrate to lottie-react library

## Problem

During tests and development, React warns about using deprecated React lifecycle methods (`componentWillUpdate`). This is due to the `react-lottie` library which is abandoned and still uses the deprecated method [[1]]. 

This is fine for now, but mildly annoying. Furthermore, the method will be removed in future React versions.

A fix would be to migrate to a more well-maintained library like `lottie-react`.

Closes issue #1083 

## Solution

Migrate to [lottie-react](https://github.com/Gamote/lottie-react) which is written in TypeScript and has generally better code quality. It is also expected to have React 17 support soon [[2]].

I considered [react-lottie-player](https://github.com/mifi/react-lottie-player) as well, but decided upon `lottie-react` since it is written in TypeScript.

[1]: https://github.com/chenqingspring/react-lottie/issues/119
[2]: https://github.com/Gamote/lottie-react/issues/29

**Improvements**:

- Removed `react-lottie` and `@types/react-lottie`
- Installed `lottie-react`

## Tests

These tests have been verified on Amplify on the `chore-migrate-react-lottie` branch, but not on the `staging` endpoint.

1. Deploy the change on staging.
2. Visit the landing page
3. The landing animation should play just fine with no errors/warning in the console.

Lottie isn't used anywhere else, so this is the only test needed.

## Deploy Notes

**New dependencies**:

- `lottie-react`

**Removed dependencies**:

- `react-lottie`
- `@types/react-lottie`
